### PR TITLE
feat: Dedicated deployments

### PR DESCRIPTION
**Pull request by arpan29**

### CODE REVIEW

**Comment by arpan29** [comment]
@gja Please have a look at the flags. I am not sure what would be a good letter for the shortcuts for each flag. Also, is there a way to replace the version of CLI in README.md, or do I need to it with Replace All in my IDE?

File: src/commands/deploy-backend.ts [hunk 1]
```diff
@@ -21,47 +36,63 @@ export default class DeployBackend extends BaseCommand {
     organizationId: flags.string({char: 'o', description: 'Organization ID', default: ''}),
     subdomain: flags.string({char: 's', description: 'Subdomain'}),
     mode: flags.string({char: 'm', description: 'Backend Mode', default: 'graphql', options: ['readonly', 'graphql', 'flexible']}),
+    type: flags.string({char: 'T', description: 'Backend Type', default: 'shared', options: ['shared', 'dedicated']}),
```
Comment (gja) [inline src/commands/deploy-backend.ts]: default slash-graphql.

File: src/commands/deploy-backend.ts [hunk 2]
```diff
@@ -21,47 +36,63 @@ export default class DeployBackend extends BaseCommand {
...
+    jaeger: flags.string({char: 'J', description: 'Enable Jaeger (Only works for dedicated backends)', default: 'false', options: ['true', 'false']}),
+    acl: flags.string({char: 'A', description: 'Enable ACL (Only works for dedicated backends)', default: 'false', options: ['true', 'false']}),
+    dgraphHA: flags.string({char: 'H', description: 'Enable High Availability (Only works for dedicated backends)', default: 'false', options: ['true', 'false']}),
+    alphaNodes: flags.string({char: 'N', description: 'Number of Alpha Nodes Needed (Only works for dedicated backends)', default: '1'}),
+    zeroNodes: flags.string({char: 'Z', description: 'Number of Zero Nodes Needed (Only works for dedicated backends)', default: '1'}),
+    size: flags.string({char: 'S', description: 'Backend Size (Only Works for dedicated backends)', default: 'small', options: ['small', 'medium', 'large', 'xlarge']}),
+    dataFile: flags.string({char: 'd', description: 'Data File Path for Bulk Loader (Only works for dedicated backends)', default: ''}),
```
Comment (gja) [inline src/commands/deploy-backend.ts]: drop these short codes

File: src/commands/deploy-backend.ts [hunk 3]
```diff
@@ -21,47 +36,63 @@ export default class DeployBackend extends BaseCommand {
     organizationId: flags.string({char: 'o', description: 'Organization ID', default: ''}),
     subdomain: flags.string({char: 's', description: 'Subdomain'}),
     mode: flags.string({char: 'm', description: 'Backend Mode', default: 'graphql', options: ['readonly', 'graphql', 'flexible']}),
+    type: flags.string({char: 'T', description: 'Backend Type', default: 'shared', options: ['shared', 'dedicated']}),
+    jaeger: flags.string({char: 'J', description: 'Enable Jaeger (Only works for dedicated backends)', default: 'false', options: ['true', 'false']}),
+    acl: flags.string({char: 'A', description: 'Enable ACL (Only works for dedicated backends)', default: 'false', options: ['true', 'false']}),
+    dgraphHA: flags.string({char: 'H', description: 'Enable High Availability (Only works for dedicated backends)', default: 'false', options: ['true', 'false']}),
+    alphaNodes: flags.string({char: 'N', description: 'Number of Alpha Nodes Needed (Only works for dedicated backends)', default: '1'}),
```
Comment (gja) [inline src/commands/deploy-backend.ts]: don't expose this yet. Lots to tweak

File: src/lib/index.ts [hunk 4]
```diff
@@ -139,6 +139,7 @@ export abstract class BaseCommand extends Command {
   }
 
   async sendGraphQLRequest(apiServer: string, token: string, query: string, variables: any) {
+    this.log(variables);
```
Comment (gja) [inline src/lib/index.ts]: remove log

**Review by gja — APPROVED** [review]

**Pull request merged**